### PR TITLE
Update controller-runtime from 0.14 to 0.15

### PR DESCRIPTION
**Pull request by bshephar**
This change updates the version of controller-runtime to v0.15.0 along with the necessary method changes to satisfy the interface changes that come with v0.15.0. This is similar to what has been done in Kubebuilder here:
https://github.com/kubernetes-sigs/kubebuilder/pull/3394/files

The specific interface changes in this version are defined in this controller-runtime commit:
https://github.com/kubernetes-sigs/controller-runtime/commit/8770b4d3b5425b156f48c971ee140ea25f9c1bfa#diff-0b6c79b3483fe5ad476ea26192f2d8bfe05a44688209721b626dc793204df28eR42

### CODE REVIEW

**Comment by bshephar** [comment]
Looks like we also need to update `helm-operator-plugins` now to this version of kubebuilder since it removes the v2-alpha directory and instead uses v2:
```
❯ go vet ./...
../../../go/pkg/mod/github.com/operator-framework/helm-operator-plugins@v0.0.12-0.20230413193425-4632388adc61/pkg/plugins/hybrid/v1alpha/scaffolds/init.go:35:2: no required module provides package sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v2-alpha; to add it:
	go get sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v2-alpha
```

https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/common/kustomize

**Comment by bshephar** [comment]
> Looks like we also need to update `helm-operator-plugins` now to this version of kubebuilder since it removes the v2-alpha directory and instead uses v2:
> 
> ```
> ❯ go vet ./...
> ../../../go/pkg/mod/github.com/operator-framework/helm-operator-plugins@v0.0.12-0.20230413193425-4632388adc61/pkg/plugins/hybrid/v1alpha/scaffolds/init.go:35:2: no required module provides package sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v2-alpha; to add it:
> 	go get sigs.k8s.io/kubebuilder/v3/pkg/plugins/common/kustomize/v2-alpha
> ```
> 
> https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/common/kustomize

Ok, so updating things in [`helm-operator-plugins`](https://github.com/operator-framework/helm-operator-plugins/pull/216) gets us to these last 3:
```
❯ go vet ./...
# k8s.io/client-go/applyconfigurations/meta/v1
../../../go/pkg/mod/k8s.io/client-go@v0.27.2/applyconfigurations/meta/v1/unstructured.go:64:38: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to proto.NewOpenAPIData
# sigs.k8s.io/kustomize/kyaml/openapi
../../../go/pkg/mod/sigs.k8s.io/kustomize/kyaml@v0.14.1/openapi/openapi.go:683:33: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to swagger.FromGnostic
# k8s.io/kubectl/pkg/util/openapi
../../../go/pkg/mod/k8s.io/kubectl@v0.27.2/pkg/util/openapi/openapi.go:52:38: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to proto.NewOpenAPIData
```

File: go.mod [hunk 1]
```diff
@@ -29,19 +29,19 @@ require (
...
+	golang.org/x/tools v0.9.1
 	gomodules.xyz/jsonpatch/v3 v3.0.1
 	helm.sh/helm/v3 v3.10.3
-	k8s.io/api v0.26.2
-	k8s.io/apiextensions-apiserver v0.26.2
-	k8s.io/apimachinery v0.26.2
+	k8s.io/api v0.27.2
+	k8s.io/apiextensions-apiserver v0.27.2
+	k8s.io/apimachinery v0.27.2
 	k8s.io/cli-runtime v0.26.2
```
[the remark this answers is not in the document]
Comment (bshephar) [reply go.mod]: Done now.

File: go.mod [hunk 2]
```diff
@@ -29,19 +29,19 @@ require (
...
-	k8s.io/client-go v0.26.2
+	k8s.io/client-go v0.27.2
 	k8s.io/kubectl v0.26.2
-	k8s.io/utils v0.0.0-20230220204549-a5ecb0141aa5
-	sigs.k8s.io/controller-runtime v0.14.5
+	k8s.io/utils v0.0.0-20230505201702-9f6742963106
+	sigs.k8s.io/controller-runtime v0.15.0
 	sigs.k8s.io/controller-tools v0.11.3
 	sigs.k8s.io/kubebuilder/v3 v3.9.1
```
[the remark this answers is not in the document]
Comment (bshephar) [reply go.mod]: Done this, although it resulted in the removal of v2-alpha, so a few extra changes across operator-framework projects were necessary as a result.

File: go.mod [hunk 3]
```diff
@@ -5,22 +5,22 @@ go 1.19
...
 	github.com/markbates/inflect v1.0.4
 	github.com/maxbrunsfeld/counterfeiter/v6 v6.2.2
-	github.com/onsi/ginkgo/v2 v2.7.0
-	github.com/onsi/gomega v1.24.2
+	github.com/onsi/ginkgo/v2 v2.9.5
+	github.com/onsi/gomega v1.27.7
 	github.com/operator-framework/api v0.17.4-0.20230223191600-0131a6301e42
 	github.com/operator-framework/helm-operator-plugins v0.0.12-0.20230413193425-4632388adc61
 	github.com/operator-framework/java-operator-plugins v0.7.1-0.20230306190439-0eed476d2b75
 	github.com/operator-framework/operator-lib v0.11.1-0.20230306195046-28cadc6b6055
```
[the remark this answers is not in the document]
Comment (bshephar) [reply go.mod]: Done now in the latest revision.

**Review by varshaprasad96 — COMMENTED** [review]
@bshephar I tried these locally. This change requires helm-operator-plugins (https://github.com/operator-framework/helm-operator-plugins/pull/216) and java-plugins (https://github.com/operator-framework/java-operator-sdk) to be bumped first.

However, meanwhile just to test if openapi bump is causing problems, I removed both the plugin dependencies, bumped KB to v3.11.0 (https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.11.0) and tested locally. Looks like `go mod tidy` works well, there are just some breaking changes which need to be addressed in the codebase. Considering this, I believe the error you are facing is because of both the plugins haven't been bumped their k8s dependencies yet. Could we merge both of them to proceed further? Thanks!

**Comment by varshaprasad96** [comment]
@bshephar Can we merge the work done here in this PR (https://github.com/operator-framework/operator-sdk/pull/6514). If you're fine, we can collaborate on this and I can cherry pick some of the commits from your branch and add it to the other PR. This is because the other one includes changes in KB and other operators that need to fixed along with this. Also, we are blocked on one of the controller-runtime's PR to include logging to the event handler.

**Comment by bshephar** [comment]
> @bshephar Can we merge the work done here in this PR (#6514). If you're fine, we can collaborate on this and I can cherry pick some of the commits from your branch and add it to the other PR. This is because the other one includes changes in KB and other operators that need to fixed along with this. Also, we are blocked on one of the controller-runtime's PR to include logging to the event handler.

Hey! Yeah, we sure can. I'm happy with that, I'll close this one.

**Pull request closed without merging**